components/README.md

```diff
@@ -1,4 +1,58 @@
 # Component Integration
 
-TODO: This document provides steps to integrate new components with
-the operator
+The `components` dir of the codebase is hosts all the component specific logic of the operator. Since, ODH operator is an
+integration point to deploy ODH component manifests it is essential to have common processes to integrate new components.
+
+## Integrating a new component
+
+To ensure a component is integrated seamlessly in the operator, follow the steps below -
+
+### Add Component to DataScienceCluster API spec
+
+DataScienceCluster CRD is responsible for defining the component fields and exposing them to end users.
+Add your component to it's [api spec](https://github.com/opendatahub-io/opendatahub-operator/blob/main/apis/datasciencecluster/v1alpha1/datasciencecluster_types.go#L40):
+
+```go
+type Components struct {
+   NewComponent newcomponent.newComponentName `json:"newcomponent,omitempty"`
+}
+```
+
+### Add Component module
+
+- Add a new module, `<newComponent>`,  under `components/` directory to define code specific to the new component. Example
+can be found [here](https://github.com/opendatahub-io/opendatahub-operator/tree/main/components/datasciencepipelines)
+- Define `Path` and `ComponentName` [variables](https://github.com/opendatahub-io/opendatahub-operator/blob/main/components/datasciencepipelines/datasciencepipelines.go#L11) for the new component.
+
+### Implement common Interface
+
+- Define struct that includes a shared struct `Component` with common fields.
+- Implement [interface](https://github.com/opendatahub-io/opendatahub-operator/blob/main/components/datasciencepipelines/datasciencepipelines.go#L39) methods according to your component
```

Comment:
it might be helpful to add method documentation into the source file (component.go) and then link it here, instead of the datasciencepipeline one...